### ISSUE TITLE
Regarding Image pre-processing.

### ISSUE THREAD

**Issue by vnalluri** [cvjena/cnn-models#9]
Hello Simon
I was able to replicate the published result of 18.1% top5 error-rate on the ILSVRC 2012 classification task using the uploaded alexnet_cvgj_iter_320000.caffemodel.
But when I tried recreating my own .caffemodel by training (2 GPUs, batch size 64 per GPU) from scratch, I was only able to achieve top5 error-rate 20.6%. One of the difference I thought could be in the image pre-processing. (Batch size 128 per GPU on 2 GPUs gave me worse results 21.4% top5 error-rate)
I have been using https://github.com/BVLC/caffe/blob/master/examples/imagenet/create_imagenet.sh to create the database for training and validation, which I believe only does resizing the Imagenet data to 256x256 images

What kind of image processing was done for the training and validation image set, for training the alexnet_cvgj model?

Thanks
Vamsi

**Comment by MarcelSimon** [comment]
Hi!
I used the ImageDataLayer instead of an LMDB in my experiments. As you correctly noticed, the LMDB creation discards the aspect ratio and warps all images to 256x256. 
In my experiments, I manually resized all images to smaller side 256px using the convert tool and used the ImageDataLayer to read these images on the fly.

**Comment by vnalluri** [comment]
Thanks for your quick replies Simon. When you say convert tool to resize the image to 256px, is this what you use - _convert "input.jpeg" -resize 256x256! -quality 100 "output.jpeg"_ (I just got this command by quick google-ing)

**Comment by MarcelSimon** [comment]
I used convert input.jpg -resize 256x256^ output.jpg , where ^ means the aspect ratio is preserved and minimum values for width and height are 256, respectively.

**Comment by MarcelSimon** [comment]
Another aspect might be the batch size. As batch normalization is used, a smaller batch size leads to stronger regularization and hence might give better results. I did not evaluate this though, it's just a feeling given you also observed an improvement when going from 128 to 64 batch size.

**Comment by sunformoon** [comment]
Hi @MarcelSimon,
Thank you for your sharing models! I am also wondering how you deal with the mean image during the pre-processing. I didn't see the "imagenet_mean.binaryproto" in your prototxt. Does it mean that you haven't considered the mean cause the data is directly followed by a BN layer?

Thank you!

**Comment by MarcelSimon** [comment]
You are right, a BN layers calculates a moving average for each channel (in addition to the variance) and normalizes the input with these averages in testing.
